packages/contracts-bedrock/test/OptimismPortal.t.sol

```diff
@@ -163,145 +163,258 @@ contract OptimismPortal_Test is Portal_Initializer {
+            _data: _data
+        });
+
+        // EOA emulation
+        vm.prank(address(this), address(this));
+        op.depositTransaction({ _to: address(0), _value: 0, _gasLimit: _gasLimit, _isCreation: true, _data: _data });
     }
 
     /// @dev Tests that `depositTransaction` succeeds for a contract
     ///      depositing a contract creation with 0 value.
-    function test_depositTransaction_createWithZeroValueForContract_succeeds() external {
-        vm.expectEmit(true, true, false, true);
-        emitTransactionDeposited(
-            AddressAliasHelper.applyL1ToL2Alias(address(this)),
-            ZERO_ADDRESS,
-            ZERO_VALUE,
-            ZERO_VALUE,
-            NON_ZERO_GASLIMIT,
-            true,
-            NON_ZERO_DATA
+    function testFuzz_depositTransaction_createWithZeroValueForContract_succeeds(
+        uint64 _gasLimit,
+        bytes memory _data
+    )
+        external
+    {
+        _gasLimit = uint64(
+            bound(_gasLimit, op.minimumGasLimit(uint64(_data.length)), systemConfig.resourceConfig().maxResourceLimit)
         );
 
-        op.depositTransaction(ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, true, NON_ZERO_DATA);
+        vm.expectEmit(address(op));
+        emitTransactionDeposited({
+            _from: AddressAliasHelper.applyL1ToL2Alias(address(this)),
+            _to: address(0),
+            _value: 0,
+            _mint: 0,
+            _gasLimit: _gasLimit,
+            _isCreation: true,
+            _data: _data
+        });
+
+        op.depositTransaction({ _to: address(0), _value: 0, _gasLimit: _gasLimit, _isCreation: true, _data: _data });
     }
 
     /// @dev Tests that `depositTransaction` succeeds for an EOA depositing a tx with ETH.
-    function test_depositTransaction_withEthValueFromEOA_succeeds() external {
-        // EOA emulation
-        vm.prank(address(this), address(this));
-
-        vm.expectEmit(true, true, false, true);
-        emitTransactionDeposited(
-            address(this), NON_ZERO_ADDRESS, NON_ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA
+    function testFuzz_depositTransaction_withEthValueFromEOA_succeeds(
+        address _to,
+        uint256 _value,
+        uint64 _gasLimit,
+        bool _isCreation,
+        bytes memory _data
+    )
+        external
+    {
+        _gasLimit = uint64(
+            bound(_gasLimit, op.minimumGasLimit(uint64(_data.length)), systemConfig.resourceConfig().maxResourceLimit)
         );
+        if (_isCreation) _to = address(0);
 
-        op.depositTransaction{ value: NON_ZERO_VALUE }(
-            NON_ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA
-        );
-        assertEq(address(op).balance, NON_ZERO_VALUE);
+        // EOA emulation
+        vm.deal(address(this), _value);
+
+        vm.expectEmit(address(op));
+        emitTransactionDeposited({
+            _from: address(this),
+            _to: _to,
+            _value: _value,
```

Comment:
Since this test is `withEthValue` should we enforce nonzero value with `_value = bound(_value, 1, type(uint256).max)`?